SA/design_concept/content/poc/architecture.typ

```diff
@@ -1,23 +1,44 @@
 = Architecture <poc-architecture>
-Content: Explain architecture with electron, threepenny, frontend & backend
+This chapter describes the architecture of the PoC application.
 
-== Translation Layer <translation-layer>
+As described in @impl_options_decision, the PoC uses Electron.js and Threepenny-GUI for its implementation.
+This choice results in the applicaiton havig a "remote" user interface, as can be seen in @container-diagram.
 
-The translation layer has been built between the UI and the inference engine
-and is responsible for converting the different data models.
+#figure(
+  image("../../static/container_diagram.png", width: 50%),
+  caption: "Container Diagram for VisualFP PoC"
+)<container-diagram>
+
+The Threepenny UI starts a local web-server where it serves static files for the UI.
+The Electron.js app itself doesn't contain any logic and just acts as a browser.
+
+Theoretically, Threepenny could also be used to host a regular web application, eliminating the need for an Electron app.
+However, since Threepenny frequently controls the displaying browser via web sockets, it is recommended to avoid this, reducing the latency for UI updates.
+@threepenny_gui
+
+The Threepenny application itself consists of three components, which can be seen in @component-diagram.
```

Comment:
I suggest adding a citation. I think the original author is Simon Brown, and he has written this book: https://leanpub.com/visualising-software-architecture

If you've already made the effort to make it correct, so we might as well take credit for it 😊

Reply:
I'm not sure if that actually qualifies as a citation. E.g. if you make an UML class diagram you don't add a citation for UML. However, I'll add C4 to the diagram legend.